### PR TITLE
[Hub Generated] Review request for Microsoft.AlertsManagement to add version preview/2019-05-05-preview

### PR DESCRIPTION
**Pull request by roramc**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/app/branch/roramc/azure-rest-api-specs/dev-alertsmanagement-Microsoft.AlertsManagement-2019-05-05-preview...master).

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by veronicagg — CHANGES_REQUESTED** [review]
@roramc there looks to be a syntax error in the swagger file, could you check https://dev.azure.com/azure-sdk/public/_build/results?buildId=21433&view=logs&jobId=5ea8264b-28a6-50c8-6f1c-6d360776db7e&taskId=cb0be435-f7f1-5f7a-37e4-9cbab5376524&lineStart=19&lineEnd=20&colStart=1&colEnd=1 ?

**Comment by veronicagg** [comment]
Tagging @KrisBash who was looking at the original PR from ARM side.

**Comment by roramc** [comment]
> @roramc there looks to be a syntax error in the swagger file, could you check https://dev.azure.com/azure-sdk/public/_build/results?buildId=21433&view=logs&jobId=5ea8264b-28a6-50c8-6f1c-6d360776db7e&taskId=cb0be435-f7f1-5f7a-37e4-9cbab5376524&lineStart=19&lineEnd=20&colStart=1&colEnd=1 ?

fixed the syntax issue/model validation, etc.

**Comment by veronicagg** [comment]
Please review errors at https://portal.azure-devex-tools.com/app/branch/roramc/azure-rest-api-specs/dev-alertsmanagement-Microsoft.AlertsManagement-2019-05-05-preview...master/linter particularly the ones under "IsNew" column = true. 
Property names like https://github.com/roramc/azure-rest-api-specs/blob/48467c63614bb04de4535ee9da6db3a232089714/specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-05-05-preview/AlertsManagement.json#L1687 should be camel cased without spaces.
Please check error generating Python SDK https://dev.azure.com/azure-sdk/public/_build/results?buildId=22034&view=logs&jobId=957f396e-b28a-53c4-b5bb-b8c6d959bd2d&taskId=142feabb-6d02-5968-caa1-f5b3c04c5cae&lineStart=136&lineEnd=137&colStart=1&colEnd=1

**Review by veronicagg — CHANGES_REQUESTED** [review]
minor: we generally use 2 spaces for indentation.  See pointer to other CI errors above.

**Comment by roramc** [comment]
> Please review errors at https://portal.azure-devex-tools.com/app/branch/roramc/azure-rest-api-specs/dev-alertsmanagement-Microsoft.AlertsManagement-2019-05-05-preview...master/linter particularly the ones under "IsNew" column = true.
> Property names like https://github.com/roramc/azure-rest-api-specs/blob/48467c63614bb04de4535ee9da6db3a232089714/specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-05-05-preview/AlertsManagement.json#L1687 should be camel cased without spaces.
> Please check error generating Python SDK https://dev.azure.com/azure-sdk/public/_build/results?buildId=22034&view=logs&jobId=957f396e-b28a-53c4-b5bb-b8c6d959bd2d&taskId=142feabb-6d02-5968-caa1-f5b3c04c5cae&lineStart=136&lineEnd=137&colStart=1&colEnd=1

I have fixed most of the lint errors, the camel case error are for the contract of an existing resource type and existing version. So I can't change those properties.  Also, we are just adding one new resource type for this version, should I remove the other existing resource types from the swagger and examples?

**Comment by veronicagg** [comment]
@roramc please follow up with the ARM team on signing off for this PR, since they haven't replied here yet. Regarding your question on removing existing resource types from swagger and examples, if they are part of the service you're specifying they should remain there, why would you remove them? thanks!

**Comment by roramc** [comment]
> @roramc please follow up with the ARM team on signing off for this PR, since they haven't replied here yet. Regarding your question on removing existing resource types from swagger and examples, if they are part of the service you're specifying they should remain there, why would you remove them? thanks!

The other resource types are already in GA. So we were thinking of just having the new resource type with the preview version. Not sure what is the convention around this.

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by veronicagg** [comment]
@roramc have you reviewed linter warnings and errors? 
For example:

1.  https://dev.azure.com/azure-sdk/public/_build/results?buildId=23706&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d&taskId=923b5a90-e462-5994-c166-86509c6bf5f2&lineStart=902&lineEnd=903&colStart=1&colEnd=1 should the operation be marked as x-ms-pageable?

2. https://dev.azure.com/azure-sdk/public/_build/results?buildId=23706&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d&taskId=923b5a90-e462-5994-c166-86509c6bf5f2&lineStart=906&lineEnd=907&colStart=1&colEnd=1 should the operation name be updated?

3. https://dev.azure.com/azure-sdk/public/_build/results?buildId=23706&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d&taskId=923b5a90-e462-5994-c166-86509c6bf5f2&lineStart=918&lineEnd=919&colStart=1&colEnd=1 should this name be updated?

4. https://dev.azure.com/azure-sdk/public/_build/results?buildId=23706&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d&taskId=923b5a90-e462-5994-c166-86509c6bf5f2&lineStart=954&lineEnd=955&colStart=1&colEnd=1 did you consider flattening?

**Comment by roramc** [comment]
> @roramc have you reviewed linter warnings and errors?
> For example:
> 
> 1. https://dev.azure.com/azure-sdk/public/_build/results?buildId=23706&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d&taskId=923b5a90-e462-5994-c166-86509c6bf5f2&lineStart=902&lineEnd=903&colStart=1&colEnd=1 should the operation be marked as x-ms-pageable?
> 2. https://dev.azure.com/azure-sdk/public/_build/results?buildId=23706&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d&taskId=923b5a90-e462-5994-c166-86509c6bf5f2&lineStart=906&lineEnd=907&colStart=1&colEnd=1 should the operation name be updated?
> 3. https://dev.azure.com/azure-sdk/public/_build/results?buildId=23706&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d&taskId=923b5a90-e462-5994-c166-86509c6bf5f2&lineStart=918&lineEnd=919&colStart=1&colEnd=1 should this name be updated?
> 4. https://dev.azure.com/azure-sdk/public/_build/results?buildId=23706&view=logs&jobId=688669d0-441c-57c3-cf6d-f89a22ccfa5d&taskId=923b5a90-e462-5994-c166-86509c6bf5f2&lineStart=954&lineEnd=955&colStart=1&colEnd=1 did you consider flattening?

fixed all the above and few more as well for the new resource type "actionRules". The remaining are for the old resource types.

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-05-05-preview/AlertsManagement.json [hunk 1]
```diff
@@ -337,8 +337,7 @@
           "200": {
             "description": "OK. Returns list of all smartGroups.",
             "schema": {
-              "$ref": "#/definitions/smartGroupsList",
-              "description": "List of smart groups in value property."
+              "$ref": "#/definitions/smartGroupsList"
```
Comment (veronicagg) [inline specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-05-05-preview/AlertsManagement.json]: names for model definitions should be capitalized for consistency, I noticed you have some starting with lower case and some upper case in the "definitions" section.

**Review by veronicagg — CHANGES_REQUESTED** [review]
left one more question inline, rest looks good. thanks!

File: specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-05-05-preview/AlertsManagement.json (hunk 1 above)
Comment (roramc) [reply specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-05-05-preview/AlertsManagement.json]: Done

**Review by veronicagg — APPROVED** [review]
@roramc  Thanks for addressing the comments!

**Pull request merged**